components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -2522,6 +2537,84 @@ public String createApplication(ServiceProvider application, String tenantDomain
         }
         return resourceId;
     }
+    
+    /**
+     * Create the application. This method creates the service provider with protocol configurations.
+     *
+     * @param applicationModel ApplicationModelDTO containing the app information
+     * @param tenantDomain     Tenant domain.
+     * @param username         Username.
+     * @return Application ID.
+     * @throws IdentityApplicationManagementException Identity Application Management Exception.
+     */
+    public String createApplication(ApplicationDTO applicationModel, String tenantDomain, String username)
+            throws IdentityApplicationManagementException {
+
+        List<InboundAuthenticationRequestConfig> addedInbounds = new ArrayList<>();
+        try {
+            ServiceProvider application = applicationModel.getServiceProvider();
+             addedInbounds = addInboundAuthenticationProtocolsToApplication(
+                    application, applicationModel.getInboundProtocolConfigurationDto());
+            
+            return createApplication(application, tenantDomain, username);
+        } catch (Throwable e) {
+            /*
+             * For more information read https://github.com/wso2/product-is/issues/12579. This is to overcome the
+             * above issue.
+             */
+            if (log.isDebugEnabled()) {
+                log.debug("Server encountered unexpected error. Rolling back created application data.", e);
```

Comment:
We don't need the `isDebugEnabled` check

Reply:
Fixed with 51b85af647109d22ee800443e12e61cc0b09c1dc